app/src/main/java/com/github/se/orator/ui/offline/OfflineRecordingScreen.kt

```diff
@@ -4,93 +4,140 @@ import android.Manifest
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.*
-import androidx.compose.material3.Button
-import androidx.compose.material3.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Mic
+import androidx.compose.material3.*
 import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.se.orator.R
+import com.github.se.orator.model.symblAi.SpeakingViewModel
 import com.github.se.orator.ui.navigation.NavigationActions
-import com.github.se.orator.ui.speaking.OfflineSpeakingViewModel
+import com.github.se.orator.ui.navigation.Screen
 
-/**
- * OfflineRecordingScreen is a composable screen for offline mode. It allows the user to record
- * audio locally and play it back.
- *
- * @param navigationActions Provides navigation actions for navigating between screens.
- * @param question The question text selected by the user for offline practice.
- * @param offlineSpeakingViewModel ViewModel for managing audio recording and playback
- *   functionality.
- */
 @SuppressLint("MissingPermission")
 @Composable
 fun OfflineRecordingScreen(
     navigationActions: NavigationActions,
-    question: String, // Accept question as a parameter
-    offlineSpeakingViewModel: OfflineSpeakingViewModel =
-        viewModel(factory = OfflineSpeakingViewModelFactory(LocalContext.current))
+    question: String,
+    speakingViewModel: SpeakingViewModel = viewModel()
 ) {
-  // Local state to track whether recording is in progress
-  var isRecording by remember { mutableStateOf(false) }
-
-  // Local state to track if the recording permission has been granted
   var permissionGranted by remember { mutableStateOf(false) }
-
-  // Set up permission launcher for requesting microphone access
   val permissionLauncher =
-      rememberLauncherForActivityResult(
-          contract = ActivityResultContracts.RequestPermission(),
-          onResult = { isGranted -> permissionGranted = isGranted })
+      rememberLauncherForActivityResult(contract = ActivityResultContracts.RequestPermission()) {
+          isGranted ->
+        permissionGranted = isGranted
+        if (isGranted) {
+          speakingViewModel.onMicButtonClicked(true)
+        }
+      }
 
-  // Request recording permission on initial composition
-  LaunchedEffect(Unit) { permissionLauncher.launch(Manifest.permission.RECORD_AUDIO) }
+  val isRecording by speakingViewModel.isRecording.collectAsState()
+  val infiniteTransition = rememberInfiniteTransition()
```

Comment:
We should probably use the same Visualizer for the recording action for both online and offline mode. Here is the code snippet from the method we use in SpeakingScreen: 

"/**
 * A composable that visualizes audio amplitudes as a waveform.
 *
 * @param amplitudes A list of amplitude values to visualize.
 */
@Composable
fun AudioVisualizer(amplitudes: List<Float>) {
  Canvas(
      modifier =
          Modifier.fillMaxWidth()
              .height(AppDimensions.visualizerHeight)
              .testTag("audio_visualizer")) {
        val width = size.width
        val height = size.height
        val barWidth = width / amplitudes.size
        amplitudes.forEachIndexed { index, amplitude ->
          val barHeight = (amplitude / Short.MAX_VALUE) * height
          drawLine(
              color = AppColors.primaryColor,
              start = Offset(x = index * barWidth, y = height / 2 - barHeight / 2),
              end = Offset(x = index * barWidth, y = height / 2 + barHeight / 2),
              strokeWidth = barWidth)
        }
      }". This method make the visualization dependent on the user's voice, and adds UI consistency to our app!